### PR TITLE
made encryption config work

### PR DESCRIPTION
**Pull request by Alvsch**
Made the encryption option in the config actually turn on and off encryption.

Also cleaned up some of the encryption response handling.

### CODE REVIEW

File: pumpkin/src/client/client_packet.rs [hunk 1]
```diff
@@ -106,10 +106,23 @@ impl Client {
...
                 .await;
+            } else {
+                if ADVANCED_CONFIG.packet_compression.enabled {
+                    self.enable_compression().await;
+                }
+                self.finish_login(
+                    gameprofile
+                        .as_ref()
+                        .expect("The GameProfile was just set to Some, this should never fail"),
+                )
```
Comment (Snowiiii) [inline pumpkin/src/client/client_packet.rs:123]: You could actually remove the need for an optional gameprofile here. Above a new gameprofile is creates, you can store it in a new var
```rust
            let profile = GameProfile {
                id: login_start.uuid,
                name: login_start.name,
                properties: vec![],
                profile_actions: None,
            };
            *gameprofile = Some(profile);
            
             self.finish_login(
                   profile,
                )
 ```

Comment (Alvsch) [reply pumpkin/src/client/client_packet.rs:123]: borrow checker would not like this as we move profile into gameprofile

Comment (Snowiiii) [reply pumpkin/src/client/client_packet.rs:123]: `finish_login` takes in a reference, means you also can set the gameprofile below the function

**Comment by Snowiiii** [comment]
Looks good then, We should keep in mind that encryption is needed when online mode is enabled. Otherwise its Optional

**Pull request merged**